AndroidPubSub/AuthenticatedPubsub.md

```diff
@@ -0,0 +1,85 @@
+## Using AWS IOT with user sign-in
+
+The sample demonstrates enabling guest users to use the AWS IoT APIs to securely publish-to and subscribe-from MQTT topics. Following steps demonstrates usage of AWS IoT with user sign-in. 
+
+1. Follow the steps [here](https://aws-amplify.github.io/docs/android/authentication) to add authentication UI to your app.
+
+2. Attach the following IAM Policy to the role of authenticated pool to authenticate and authorize the Cognito user to communicate with AWS IoT
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "iot:AttachPrincipalPolicy",
+                "iot:Connect",
+                "iot:Publish",
+                "iot:Subscribe",
+                "iot:Receive"
+            ],
+            "Resource": [
+                "*"
+            ]
+        }
+    ]
+}
+```
+
+3. Get the credentials provider
+
+```
+credentialsProvider =
+               IdentityManager.getDefaultIdentityManager().getCredentialsProvider();
+```
+
+4. In the app, after the user logs in using the sign-in UI, get CognitoIdentity from credentialsProvider and use it to attach principal policy to the IotAndroidClient. Create IoT policy as specified in step 4 of [Using the sample](https://github.com/awslabs/aws-sdk-android-samples/tree/master/AndroidPubSub#using-the-sample) section above.
+
+```
+// Setup the logins map
+IdentityManager identityManager = IdentityManager.getDefaultIdentityManager();
+Map<String, String> logins = new HashMap();
+IdentityProvider currentIdentityProvider = identityManager.getCurrentIdentityProvider();
+String token = currentIdentityProvider.getToken();
+String providerKey = currentIdentityProvider.getCognitoLoginKey();
+logins.put(providerKey, token);
+
+AmazonCognitoIdentity cognitoIdentity;
+GetIdRequest getIdReq;
+GetIdResult getIdRes;
+cognitoIdentity = new AmazonCognitoIdentityClient(credentialsProvider);
+getIdReq = new GetIdRequest();
+getIdReq.setLogins(logins);
+getIdReq.setIdentityPoolId(COGNITO_POOL_ID); //Identity pool Id generated by amplify cli
+new Thread(new Runnable() {
+    @Override
+    public void run() {
+	getIdRes = cognitoIdentity.getId(getIdReq);
+	AttachPrincipalPolicyRequest attachPolicyReq = new AttachPrincipalPolicyRequest();
+	attachPolicyReq.setPolicyName(AWS_IOT_POLICY_NAME);
+	attachPolicyReq.setPrincipal(getIdRes.getIdentityId());
+	AWSIotClient mIotAndroidClient = new AWSIotClient(credentialsProvider);
+	mIotAndroidClient.setRegion(Region.getRegion(MY_REGION));
+	mIotAndroidClient.attachPrincipalPolicy(attachPolicyReq);
+    }
+}).start();
+```
+
+An Amazon Cognito authenticated user needs two policies to access AWS IoT. The first policy is attached to the role of the authenticated pool to authenticate and authorize the Cognito user to communicate with AWS IoT. The second policy is attached to the authenticated Cognito user ID principal for fine-grained permissions.
```

Comment:
Move this up to step 2 so that you can discuss all the policy related stuff there.